### PR TITLE
Fix #1338. Register Rider action in Solution context group

### LINKED ISSUE #1338: Exception on loading action groups and actions when running plugin in Rider
**Describe the bug**
`ext-rider.xml` duplicate action groups and actions with existing ids that are defined in main `plugin.xml`. This cause an exception when IDE tries to load actions and detects duplicates. 

**To reproduce**
1. Start Rider with Plugin installed
2. Open an AWS solution

**Expected:** All actions are loaded without exceptions.
**Actual:** Exception caused by loading same actions twice:
```
com.intellij.diagnostic.PluginException: action with the ID "aws.toolkit.serverless.template" was already registered. Action being registered is null (null); Registered action is  (null) Plugin: AWS Toolkit [Plugin: aws.toolkit]
	at com.intellij.openapi.actionSystem.impl.ActionManagerImpl.reportActionError(ActionManagerImpl.java:366)
	at com.intellij.openapi.actionSystem.impl.ActionManagerImpl.reportActionError(ActionManagerImpl.java:361)
	at com.intellij.openapi.actionSystem.impl.ActionManagerImpl.registerChameleon(ActionManagerImpl.java:1086)
	at com.intellij.openapi.actionSystem.impl.ActionManagerImpl.addToMap(ActionManagerImpl.java:1059)
	at com.intellij.openapi.actionSystem.impl.ActionManagerImpl.registerAction(ActionManagerImpl.java:1042)
	at com.intellij.openapi.actionSystem.impl.ActionManagerImpl.registerOrReplaceActionInner(ActionManagerImpl.java:661)
	at com.intellij.openapi.actionSystem.impl.ActionManagerImpl.processGroupElement(ActionManagerImpl.java:727)
	at com.intellij.openapi.actionSystem.impl.ActionManagerImpl.processActionsChildElement(ActionManagerImpl.java:1019)
	at com.intellij.openapi.actionSystem.impl.ActionManagerImpl.registerPluginActions(ActionManagerImpl.java:479)
	at com.intellij.openapi.actionSystem.impl.ActionManagerImpl.<init>(ActionManagerImpl.java:133)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
	at org.picocontainer.defaults.InstantiatingComponentAdapter.newInstance(InstantiatingComponentAdapter.java:193)
	at com.intellij.util.pico.CachingConstructorInjectionComponentAdapter.doGetComponentInstance(CachingConstructorInjectionComponentAdapter.java:88)
	at com.intellij.util.pico.CachingConstructorInjectionComponentAdapter.instantiateGuarded(CachingConstructorInjectionComponentAdapter.java:66)
	at com.intellij.util.pico.CachingConstructorInjectionComponentAdapter.getComponentInstance(CachingConstructorInjectionComponentAdapter.java:48)
	at com.intellij.openapi.components.impl.ComponentManagerImpl$ComponentConfigComponentAdapter.getComponentInstance(ComponentManagerImpl.java:481)
	at com.intellij.openapi.components.impl.ComponentManagerImpl$ComponentConfigComponentAdapter.getComponentInstance(ComponentManagerImpl.java:462)
	at com.intellij.util.pico.DefaultPicoContainer.getLocalInstance(DefaultPicoContainer.java:246)
	at com.intellij.util.pico.DefaultPicoContainer.getComponentInstance(DefaultPicoContainer.java:213)
	at org.picocontainer.defaults.BasicComponentParameter.resolveInstance(BasicComponentParameter.java:77)
	at org.picocontainer.defaults.ComponentParameter.resolveInstance(ComponentParameter.java:114)
	at com.intellij.util.pico.CachingConstructorInjectionComponentAdapter.getConstructorArguments(CachingConstructorInjectionComponentAdapter.java:109)
	at com.intellij.util.pico.CachingConstructorInjectionComponentAdapter.doGetComponentInstance(CachingConstructorInjectionComponentAdapter.java:88)
	at com.intellij.util.pico.CachingConstructorInjectionComponentAdapter.instantiateGuarded(CachingConstructorInjectionComponentAdapter.java:66)
	at com.intellij.util.pico.CachingConstructorInjectionComponentAdapter.getComponentInstance(CachingConstructorInjectionComponentAdapter.java:48)
	at com.intellij.openapi.components.impl.ServiceManagerImpl$MyComponentAdapter.createAndInitialize(ServiceManagerImpl.java:243)
	at com.intellij.openapi.components.impl.ServiceManagerImpl$MyComponentAdapter.lambda$getComponentInstance$0(ServiceManagerImpl.java:232)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:591)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:537)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:59)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeNonCancelableSection(CoreProgressManager.java:197)
	at com.intellij.openapi.components.impl.ServiceManagerImpl$MyComponentAdapter.getComponentInstance(ServiceManagerImpl.java:231)
	at com.intellij.util.pico.DefaultPicoContainer.getLocalInstance(DefaultPicoContainer.java:246)
	at com.intellij.util.pico.DefaultPicoContainer.getComponentInstance(DefaultPicoContainer.java:213)
	at org.picocontainer.defaults.BasicComponentParameter.resolveInstance(BasicComponentParameter.java:77)
	at org.picocontainer.defaults.ComponentParameter.resolveInstance(ComponentParameter.java:114)
	at com.intellij.util.pico.CachingConstructorInjectionComponentAdapter.getConstructorArguments(CachingConstructorInjectionComponentAdapter.java:109)
	at com.intellij.util.pico.CachingConstructorInjectionComponentAdapter.doGetComponentInstance(CachingConstructorInjectionComponentAdapter.java:88)
	at com.intellij.util.pico.CachingConstructorInjectionComponentAdapter.instantiateGuarded(CachingConstructorInjectionComponentAdapter.java:66)
	at com.intellij.util.pico.CachingConstructorInjectionComponentAdapter.getComponentInstance(CachingConstructorInjectionComponentAdapter.java:48)
	at com.intellij.openapi.components.impl.ComponentManagerImpl$ComponentConfigComponentAdapter.getComponentInstance(ComponentManagerImpl.java:481)
	at com.intellij.openapi.components.impl.ComponentManagerImpl.createComponents(ComponentManagerImpl.java:126)
	at com.intellij.openapi.application.impl.ApplicationImpl.lambda$load$7(ApplicationImpl.java:414)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:169)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:591)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:537)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:59)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:156)
	at com.intellij.openapi.application.impl.ApplicationImpl.load(ApplicationImpl.java:414)
	at com.intellij.idea.IdeaApplication.lambda$executeInitAppInEdt$5(IdeaApplication.java:149)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:313)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:776)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:727)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:746)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:377)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)
```

**Screenshots**
None.

**Your Environment**
```
JetBrains Rider 2019.2
Build #RD-192.5895.1068, built on August 7, 2019
Licensed to JetBrains Team / Sergey Dubov
You have a perpetual fallback license for this version
Subscription is active until December 10, 2020
Runtime version: 11.0.3+12-b304.25 x86_64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
macOS 10.15.1
GC: G1 Young Generation, G1 Old Generation
Memory: 512M
Cores: 8
Registry: ide.tree.horizontal.default.autoscrolling=false, performance.watcher.sampling.interval.ms=100, show.diff.preview.as.editor.tab=true, ide.borderless.title.product=false, performance.watcher.unresponsive.interval.ms=500, ide.tooltip.initialDelay.highlighter=0, java.completion.argument.hints.internal=false, search.everywhere.settings=true, parameter.info.max.visible.rows=10, ide.win.file.chooser.native=true, ide.borderless.title.project.path=false, actionSystem.fix.alt.gr=false, search.everywhere.pattern.checking=false, ide.tooltip.initialDelay=0, ide.borderless.title.classpath=false, ide.require.transaction.for.model.changes=false, rider.enable.designer.winForms=false
Non-Bundled Plugins: aws.toolkit
```

**Additional context**
This issue most probably need to be fixed on code level - `update()` should check the call context and disable one of actions. 
This issues could be fixed using `ActionConfigurationCustomizer` when switching to 193 Idea SDK.

### PR DESCRIPTION
**Pull request by sdubov**
## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Description
- Register a separate action in Rider in Soluiton context and disable in Project group

## Motivation and Context
Registration group and actions with same ID cause an exception when running Rider IDE with enabled plugin.

## Related Issue(s)
#1338 

## Testing
Verified in Rider and IDEA.

## Screenshots (if appropriate)
None.

## Checklist
- [x] I have read the **README** document
- [x] I have read the **CONTRIBUTING** document
- [x] Local run of `gradlew check` succeeds
- [x] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [x] All new and existing tests passed
- [ ] A short description of the change has been added to the **CHANGELOG**

## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: jetbrains-rider/src/software/aws/toolkits/jetbrains/services/lambda/actions/RiderDeployServerlessApplicationAction.kt [hunk 1]
```diff
@@ -0,0 +1,21 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.lambda.actions
+
+import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.util.PlatformUtils
+
+/**
+ * The action is duplicated for Rider to replace ProjectViewPopupMenu that adds action to the "Tools"
```
Comment (abrooksv) [inline jetbrains-rider/src/software/aws/toolkits/jetbrains/services/lambda/actions/RiderDeployServerlessApplicationAction.kt]: Oh...thats where it went....

File: jetbrains-core/resources/META-INF/ext-rider.xml [hunk 2]
```diff
@@ -30,9 +30,14 @@
...
     <actions>
-        <group>
+        <group id="aws.toolkit.serverless.template.rider">
             <add-to-group group-id="SolutionExplorerPopupMenu" anchor="last"/>
-            <reference ref="aws.toolkit.serverless.template"/>
+            <add-to-group group-id="EditorPopupMenu" anchor="last"/>
+            <add-to-group group-id="EditorTabPopupMenu" anchor="last"/>
+
+            <separator/>
+            <action id="serverless.application.deploy.rider" class="software.aws.toolkits.jetbrains.services.lambda.actions.RiderDeployServerlessApplicationAction"/>
```
Comment (abrooksv) [inline jetbrains-core/resources/META-INF/ext-rider.xml]: Do we need to do an `<unregister>`?

File: jetbrains-rider/src/software/aws/toolkits/jetbrains/services/lambda/actions/RiderDeployServerlessApplicationAction.kt (hunk 1 above)
Comment (sdubov) [reply jetbrains-rider/src/software/aws/toolkits/jetbrains/services/lambda/actions/RiderDeployServerlessApplicationAction.kt]: :)

File: jetbrains-core/resources/META-INF/ext-rider.xml (hunk 2 above)
Comment (sdubov) [reply jetbrains-core/resources/META-INF/ext-rider.xml]: Done. Unregister original deploy action in Rider action group to avoid check for Rider IDE.

**Pull request merged**